csrc/instance_norm_nvfuser_kernel.cu

```diff
@@ -0,0 +1,269 @@
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include <torch/extension.h>
+
+#include <torch/csrc/jit/codegen/cuda/kernel_cache.h>
+#include <torch/csrc/jit/codegen/cuda/ops/all_ops.h>
+
+// Hashing machinery for Params
+// Fowler–Noll–Vo hash function
+// see https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function
+template <typename Params>
+struct ParamsHash {
+  // Params must be a POD because we read out its memory
+  // contenst as char* when hashing
+  static_assert(std::is_pod<Params>::value, "Params is not POD");
+
+  size_t operator()(const Params& params) const {
+    auto ptr = reinterpret_cast<const uint8_t*>(&params);
+    uint32_t value = 0x811C9DC5;
+    for (int i = 0; i < (int)sizeof(Params); ++i) {
+      value ^= ptr[i];
+      value *= 0x01000193;
+    }
+    return (size_t)value;
+  }
+};
+
+template <typename Params>
+struct ParamsEqual {
+  // Params must be a POD because we read out its memory
+  // contenst as char* when comparing
+  static_assert(std::is_pod<Params>::value, "Params is not POD");
+
+  bool operator()(const Params& a, const Params& b) const {
+    auto ptr1 = reinterpret_cast<const uint8_t*>(&a);
+    auto ptr2 = reinterpret_cast<const uint8_t*>(&b);
+    return memcmp(ptr1, ptr2, sizeof(Params)) == 0;
+  }
+};
```

Comment:
Isn't [ATen/native/utils/ParamsHash.h](https://github.com/pytorch/pytorch/blob/29d759948ee7cd2275ea60ebd62c311408e3798b/aten/src/ATen/native/utils/ParamsHash.h) available?

Reply:
Thanks, originally was having some issues including this header when prototyping, but seems to build fine now...